### PR TITLE
use error value for bad URI so custom error handler could treat it special

### PR DESCRIPTION
**Pull request by jhump**
When a custom `ProtoErrorHandlerFunc` is used, it is given an error with `Unimplemented` gRPC code for cases where the server gets a request for an unrecognized URI path.

When no such handler is in place, it defaults to a "404 Not Found" error, which is what basically any REST client actually expect for such an error.

Unfortunately, there is not currently any way to distinguish between  "the server received an unrecognized URI path" and an actual "Unimplemented" error returned by a gRPC server handler. So it is not possible for a custom handler to turn those into "404 NotFound" errors, as a REST client would actually expect.

This change hoists the error out into an exported error value: `ErrUnknownURI`. That way, a custom error handler can check if `err == runtime.ErrUnknownURI` and choose to return a 404 for those errors (instead of "501 Not Implemented", which can be used for other errors with an "Unimplemented" gRPC code).

### CODE REVIEW

**Comment by achew22** [comment]
This seems eminently reasonable. Would you be willing to add a test or two to confirm the differentiation between a 404 not found and a 404 unimplemented?

**Comment by jhump** [comment]
@achew22, I added tests. PTAL.

File: runtime/mux_test.go [hunk 1]
```diff
@@ -279,6 +325,17 @@ func TestMuxServeHTTP(t *testing.T) {
 	}
 }
 
+func unknownPathIs404(ctx context.Context, mux *runtime.ServeMux, m runtime.Marshaler, w http.ResponseWriter, r *http.Request, err error) {
+	if err == runtime.ErrUnknownURI {
+		w.WriteHeader(http.StatusNotFound)
+	} else {
+		stat, _ := status.FromError(err)
```
Comment (johanbrandhorst) [inline runtime/mux_test.go]: You can use `status.Convert` for these cases. If we're not using a recent enough gRPC package, we should upgrade.

Comment (jhump) [reply runtime/mux_test.go]: Done. The version this repo uses (1.19.0) does indeed include that function.

**Pull request merged**